### PR TITLE
[Hotfix][StructuralMechanicsApplication] Fixing Sprism element 

### PR DESCRIPTION
**Pull request by loumalouomega**
The test had a very low tolerance and the elements were broken long time ago.

### CODE REVIEW

File: applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp [hunk 1]
```diff
@@ -369,41 +369,58 @@ void BaseSolidElement::CalculateMassMatrix(
...
     const double thickness = (dimension == 2 && r_prop.Has(THICKNESS)) ? r_prop[THICKNESS] : 1.0;
 
-    for ( IndexType point_number = 0; point_number < integration_points.size(); ++point_number ) {
-        GeometryUtils::JacobianOnInitialConfiguration(
-            r_geom, integration_points[point_number], J0);
-        const double detJ0 = MathUtils<double>::DetMat(J0);
-        const double integration_weight =
-            GetIntegrationWeight(integration_points, point_number, detJ0) * thickness;
-        const Vector& rN = row(Ncontainer,point_number);
+    const bool compute_lumped_mass_matrix =  rCurrentProcessInfo.Has(COMPUTE_LUMPED_MASS_MATRIX) ? rCurrentProcessInfo[COMPUTE_LUMPED_MASS_MATRIX] : false;
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp:383]: In the beams USE_CONSISTENT_MASS_MATRIX is used for this
-> there the lumped one is used by default

I think we should agree on sth and be consistent 

The solids used the consistent one so far?

**Comment by loumalouomega** [comment]
Indeed, the solid only used the consistent, this add the possibility to use
lumped matrix.

I assume that the beam used the lumped by default because it is used in
explicit schemes.

**Comment by loumalouomega** [comment]
Can anyone approve this PR?

**Comment by philbucher** [comment]
I will review the PRs where you requested me tmr ok?

**Comment by loumalouomega** [comment]
Sure, thanks a lot

File: applications/StructuralMechanicsApplication/custom_elements/solid_shell_element_sprism_3D6N.cpp [hunk 2]
```diff
@@ -1617,47 +1517,76 @@ void SolidShellElementSprism3D6N::Initialize()
 {
     KRATOS_TRY;
 
-    const GeometryType::IntegrationPointsArrayType& integration_points = GetGeometry().IntegrationPoints( mThisIntegrationMethod );
+    mFinalizedStep = true; // the creation is out of the time step, it must be true
+
+    if( GetProperties().Has(INTEGRATION_ORDER) ) {
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_elements/solid_shell_element_sprism_3D6N.cpp:1522]: isnt this in the baseclass now?

File: applications/StructuralMechanicsApplication/custom_elements/solid_shell_element_sprism_3D6N.cpp [hunk 3]
```diff
@@ -2283,6 +2222,14 @@ void SolidShellElementSprism3D6N::ComputeLocalDerivatives(
     LocalDerivativePatch(4, 1) =   0.0;
     LocalDerivativePatch(5, 1) =   L_2;
 
+//     /* Derivative in direction zeta */
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_elements/solid_shell_element_sprism_3D6N.cpp:2225]: leftover?

**Review by philbucher — APPROVED** [review]
ok

File: applications/StructuralMechanicsApplication/custom_elements/solid_shell_element_sprism_3D6N.cpp (hunk 2 above)
Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/custom_elements/solid_shell_element_sprism_3D6N.cpp:1522]: Yes, but I use the EXTENDED_GI_GAUSS, so I need to rewrite it

File: applications/StructuralMechanicsApplication/custom_elements/solid_shell_element_sprism_3D6N.cpp (hunk 3 above)
Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/custom_elements/solid_shell_element_sprism_3D6N.cpp:2225]: Well, there are two ways of compute it, I like this more

**Comment by loumalouomega** [comment]
Thanks

**Pull request merged**